src/extensionApi.ts

```diff
@@ -162,18 +154,23 @@ export class CommandHandler {
     }
 
     public async removeDeployment(context?: Protocol.DeployableState): Promise<Protocol.Status> {
+        let serverId: string;
+        let deploymentId: string;
         if (context === undefined) {
-            return Promise.reject('Please select a deployment from the Servers view to run this action.');
-        }
+            serverId = await this.selectServer('Select server to remove deployment from');
+            if (!serverId) return null;
 
-        const serverId: string = context.server.id;
-        const deploymentId: string = context.reference.label;
+            const deployables = this.serversData.serverStatus.get(serverId).deployableStates.map(value => {
+                return value.reference.label;
+            });
+            deploymentId = await vscode.window.showQuickPick(deployables, { placeHolder: 'Select deployment to remove' });
```

Comment:
should check for deploymentId to be undefined as it is when you cancel the quickpick. Otherwise one runs into this error:
![image](https://user-images.githubusercontent.com/25126/57227716-5dfb8e00-7012-11e9-8105-db223baa53eb.png)